core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -768,6 +790,12 @@ private void requestTransactions(Set<Integer> segments, Map<Address, Set<Integer
                log.failedToRetrieveTransactionsForSegments(segments, cacheName, source, e);
                // The primary owner is still in the cluster, so we can't exclude it - see ISPN-4091
                failed = true;
+               // FIXME Temporary hack to limit the number of retries when the source has a newer topology
```

Comment:
Oops, it was meant to make the failure easier to reproduce, it wasn't supposed to make it in the PR :)